vipe/graphviz/dot.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2013-2015 University of Warsaw
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__author__ = "Mateusz Kobos mkobos@icm.edu.pl"
+
+import io.StringIO
+import re
+
+whitespace_pattern = re.compile(r'\s+')
+
+# class LabelsRegister:
+#     """
+#     Register that assigns and stores mapping between user-readable labels and 
+#     technical identifiers used in the *.dot file.
+#     """
+#     def __init__(self):
+#         self.__d = {}
+#         self.__largest = 0
+# 
+#     def get(self, label):
+#         """@return node identifier assigned to given label"""
+#         if label not in self.__d:
+#             self.__largest = self.__largest+1
+#             self.__d[label] = self.__largest
+#         index = self.__d[label]
+#         return 'n'+str(index)
+    
+class DotProcessor:
+    """Takes GraphViz's graph in dot format and generates other stuff."""
+    
+    def __init__(self, dot_graph):
+        """
+        Args:
+            dot_graph (string): description of graph in GraphViz's dot format
+        """
+        self.__dot_graph = dot_graph
+    
+    def get_image_map(self):
+        """
+        Returns:
+            string: client-size image map ready to be embedded in 
+                HTML document. This is GraphViz's `cmapx` output 
+                (see http://www.graphviz.org/doc/info/output.html#a:cmapx).
+        """
+        raise NotImplementedError
+    
+    def save_image(self, output_path):
+        """Save the image to a file available at given path
+        
+        Args:
+            output_path (string): path to output png file with the graph
+        """
+        raise NotImplemented
```

Comment:
Should I assume `NotImplemented` was intentionally picked instead of `NotImplementedError`?